compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/stmt.rs

```diff
@@ -150,7 +151,21 @@ macro_rules! stmt {
 impl Stmt {
     /// `lhs = rhs;`
     pub fn assign(lhs: Expr, rhs: Expr, loc: Location) -> Self {
-        assert_eq!(lhs.typ(), rhs.typ());
+        //Temporarily work around https://github.com/model-checking/rmc/issues/95
+        //by disabling the assert and soundly assigning nondet
+        //assert_eq!(lhs.typ(), rhs.typ());
+        if lhs.typ() != rhs.typ() {
+            debug!(
+                "WARNING: assign statement with unequal types lhs {:?} rhs {:?}",
+                lhs.typ(),
+                rhs.typ()
+            );
+            let assert_stmt =
+                Stmt::assert_false("Reached assignment statement with unequal types", loc.clone());
```

Comment:
Its useful to print the types here

Reply:
`&format!("Reached assignment statement with unequal types {:?} {:?}", lhs.typ(), rhs.typ())`